codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowControllerTest.java

```diff
@@ -872,10 +873,10 @@ public void subTreeBytesShouldBeCorrect() throws Http2Exception {
 
         // Send a bunch of data on each stream.
         final IntObjectMap<Integer> streamSizes = new IntObjectHashMap<Integer>(4);
-        streamSizes.put(STREAM_A, 400);
-        streamSizes.put(STREAM_B, 500);
-        streamSizes.put(STREAM_C, 600);
-        streamSizes.put(STREAM_D, 700);
+        streamSizes.put(STREAM_A, (Integer) 400);
+        streamSizes.put(STREAM_B, (Integer) 500);
```

Comment:
@normanmaurer yes it fails to compile. My understanding of what happens is: the compiler searches for `put(int, int)`. Can't find it. Now it falls back to auto boxing. Both `put(Integer, Integer)` and `put(int, Integer)` are possible due to auto boxing  - thus the call is ambiguous. If one casts to `Integer` however, the compiler can immediately find `put(int, Integer)` and doesn't use auto boxing at all.